src/allocation_engine/interval_tree.rs

```diff
@@ -373,7 +373,7 @@ impl InnerNode {
         // (2^50 - 1) memory slots. Considering the imposed maximum height the
         // recursion is safe to use.
         if (self.height + 1) > 50 {
-            return Err(Error::ResourceExhausted);
+            return Err(Error::ResourceNotAvailable);
```

Comment:
nit: I think `Overflow` might be a better error here because we cannot insert it because the tree height is too big.

Reply:
Changed it to Overflow.